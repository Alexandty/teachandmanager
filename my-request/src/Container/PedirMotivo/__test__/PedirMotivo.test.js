import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Motivo } from "../index";

configure({ adapter: new Adapter() });

describe('Test para PerdirMotivo', () => {
    const handle = jest.fn();
    it('Debe cargar el componente', () => {
        const wrapper = shallow(<Motivo handleSubmit={handle} pristine={true} mostrar={true} titulo={'Test 1'} />);
    })
    it('Debe cargar el boton', () => {
        const wrapper = shallow(<Motivo handleSubmit={handle} pristine={true} mostrar={true} titulo={'Test 2'} />)
            .find({ bsStyle: 'success' });
        expect(wrapper.exists()).toBeTruthy();
    })
    it('Debe cargar el textarea', () => {
        const wrapper = shallow(<Motivo handleSubmit={handle} pristine={true} mostrar={true} titulo={'Test 2'} />)
            .find({ type: 'textarea' });
        expect(wrapper.exists()).toBeTruthy();
    })
    it('Validando toMatchSnapshot', () => {
        const props = {
            enviarMotivo: jest.fn(),
            handleSubmit: jest.fn(),
            pristine: true
        };
        const wrap = shallow(<Motivo {...props} />);
        expect(toJson(wrap)).toMatchSnapshot();
    });
})