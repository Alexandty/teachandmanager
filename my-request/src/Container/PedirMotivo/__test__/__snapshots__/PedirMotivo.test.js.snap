// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para PerdirMotivo Validando toMatchSnapshot 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    />
  </ModalHeader>
  <Form>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Field
        component={[Function]}
        name="motivoIngresado"
        style={
          Object {
            "width": "506px",
          }
        }
        type="textarea"
      />
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="success"
        disabled={true}
        type="submit"
      >
        Enviar
      </Button>
    </ModalFooter>
  </Form>
</Modal>
`;
